2025 02 20 - Release Notes (#4812)
Azure/AKS: CHANGELOG.md

# Azure Kubernetes Service Changelog

## Release 2025-02-20

Monitor the release status by region at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled `v20250220`.

### Announcements
* HTTP Application Routing (preview) is going to be [retired on March 3, 2025](https://azure.microsoft.com/updates?id=retirement-http-application-routing-addon-preview-for-aks-will-retire-03032025) and AKS will start to block new cluster creation with HTTP Application Routing (preview) enabled. Affected clusters must migrate to the generally available Application Routing add-on prior to that date. Refer to the [migration guide](https://learn.microsoft.com/en-us/azure/aks/app-routing-migration) for more information.

### Release Notes
* Features:
   * Application routing add-on support for [configuring the default NGINX ingress controller visibility](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration?tabs=azurecli#control-the-default-ingress-controller-configuration-when-creating-the-cluster) is now generally available in API 2025-02-01.

>    * AKS Kubernetes version 1.32 is now being rolled out and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.

Author: ```suggestion
   * 
```

   * AKS Kubernetes patch versions 1.29.12, 1.29.13, 1.30.8, 1.30.9, 1.31.4, and 1.31.5 are now available.
   * 

* Preview Features:

> 

Author: ```suggestion
   * AKS Kubernetes version 1.32 is now being rolled out in preview and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.
   * You can use the `EnableCiliumNodeSubnet' feature in preview to [create Cilium node subnet clusters](https://learn.microsoft.com/azure/aks/azure-cni-powered-by-cilium#option-3-assign-ip-addresses-from-the-node-subnet-preview) using Azure CNI Powered by Cilium.
```

# Azure Kubernetes Service Changelog

## Release 2025-02-20

Monitor the release status by region at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled `v20250220`.

### Announcements
* HTTP Application Routing (preview) is going to be [retired on March 3, 2025](https://azure.microsoft.com/updates?id=retirement-http-application-routing-addon-preview-for-aks-will-retire-03032025) and AKS will start to block new cluster creation with HTTP Application Routing (preview) enabled. Affected clusters must migrate to the generally available Application Routing add-on prior to that date. Refer to the [migration guide](https://learn.microsoft.com/en-us/azure/aks/app-routing-migration) for more information.

### Release Notes
* Features:
   * Application routing add-on support for [configuring the default NGINX ingress controller visibility](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration?tabs=azurecli#control-the-default-ingress-controller-configuration-when-creating-the-cluster) is now generally available in API 2025-02-01.

> 

Reviewer: nit: extra empty line can be removed

>    * AKS Kubernetes version 1.32 is now being rolled out and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.

Reviewer: 1.32 is in preview, so this should be under the preview features section

   * AKS Kubernetes patch versions 1.29.12, 1.29.13, 1.30.8, 1.30.9, 1.31.4, and 1.31.5 are now available.
   * 

* Preview Features:
   * AKS Kubernetes version 1.32 is now in preview.

* Bug Fixes:

>    * Resolved an issue with Istio service mesh add-on where having multiple operations with the Lua EnvoyFilter was not allowed.

Reviewer: which operations? @biefy ^

Reviewer 2: Also the Github link for issue wherever applicable pls.

   *  

* Behavior Changes:

>    * Starting in the Karpenter v0.6 image, Node Auto Provisioning (NAP) will bootstrap nodes from Scriptless to AgentBaker using custom data provided by Node Provisioning Service (NPS) and improve the stability of NAP.

Reviewer: is this user facing? what's the **behavior** change for the user?

Author: Ack, confirmed this is an internal change and should be removed

> * Component Updates:

Reviewer: vhd notes are missing right now. can we add those?

Author: VHD notes added

>    * Beginning in Kubernetes 1.32, AKS uses CNS to install Azure CNI NodeSubnet by default, meaning users will now see the CNS daemonset in their environments with no changes to functionality or required action.

Reviewer: is this a behavior change?

>    * ExtensionManager has been upgraded to  [v1.22.4](https://github.com/azure-core/ClusterConfigurationAgent/releases/tag/v1.22.4).

Reviewer: this repo is private plus extension manager is being moved to AKS service hosted components, we should remove this line

>    * ExtensionManager has been upgraded to  [v1.22.4](https://github.com/azure-core/ClusterConfigurationAgent/releases/tag/v1.22.4).

Author: ```suggestion
```

   * Node Auto Provisioning has been upgraded to use Karpenter [v0.7.2](https://github.com/Azure/karpenter-provider-azure/releases/tag/v0.7.2).
   * Updated LTS patch version 1.27.102 for Command Injection affecting Windows nodes to address [CVE-2024-9042](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-9042).
   * Updated the Retina basic image to v0.0.25 for Linux and Windows to address [CVE-2025-23047](https://nvd.nist.gov/vuln/detail/CVE-2025-23047) and [CVE-2024-45338](https://nvd.nist.gov/vuln/detail/CVE-2024-45338).
   * Updated the cost-analysis-agent image from v0.0.20 to v0.0.21. Upgrades the following dependencies in cost-analysis-agent to fix [CVE-2024-45341](https://nvd.nist.gov/vuln/detail/CVE-2024-45341) and [CVE-2024-45336](https://nvd.nist.gov/vuln/detail/CVE-2024-45336):
      * github.com/Azure/azure-sdk-for-go/sdk/azcore v1.15.0 to v1.17.0
      * github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.0 to v1.8.1
      * github.com/prometheus/common v0.60.0 to v0.62.0
      * github.com/samber/lo v1.47.0 to v1.49.1

>       * github.com/stretchr/testify v1.9.0 to v1.10.0

Reviewer: can we add links for these?

   * addon-resizer image v1.8.23-2 has been built using 
    the Dalec framework, tested and published to MCR under [oss/v2 path](mcr.microsoft.com/oss/v2/kubernetes/autoscaler/addon-resizer:v1.8.23-2). All AKS clusters will use v1.8.23-2 addon-resizer image version.

>    * Updated Azure File CSI driver liveness probe to v2.15.0 and CSI node driver registrar to v2.13.0 on AKS 1.31 and 1.32.

Reviewer: can we add link to the release?

Author: Waiting on links from Allen Wen

Author: Andy Zhang confirmed this is not a user facing change, so will remove this update

# Azure Kubernetes Service Changelog

## Release 2025-02-20

Monitor the release status by region at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled `v20250220`.

### Announcements

> * 

Reviewer 3: ```suggestion
* Using the GPU VHD image (preview) to provision GPU-enabled AKS nodes was retired on January 10, 2025 and AKS will block creation of new node pools with the GPU VHD image (preview). Follow the detailed steps to [create GPU-enabled node pools](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#use-the-aks-gpu-image-preview) using the alternative supported options.
```

> * 

Reviewer 3: ```suggestion
* HTTP Application Routing (preview) is going to be [retired on March 3, 2025](https://azure.microsoft.com/updates?id=retirement-http-application-routing-addon-preview-for-aks-will-retire-03032025) and AKS will start to block new cluster creation with HTTP Application Routing (preview) enabled. Affected clusters must migrate to the generally available Application Routing add-on prior to that date. Refer to the [migration guide](https://learn.microsoft.com/en-us/azure/aks/app-routing-migration) for more information.
```

### Release Notes

> * Features:

Reviewer 3: ```suggestion
* Features:
   * Application routing add-on support for [configuring the default NGINX ingress controller visibility](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration?tabs=azurecli#control-the-default-ingress-controller-configuration-when-creating-the-cluster) is now generally available in API 2025-02-01.

```

>    * AKS Kubernetes patch versions 1.29.12, 1.29.13, 1.30.8, 1.30.9, 1.31.4, and 1.31.5 are now available.

Author: ```suggestion
   * AKS Kubernetes version 1.32 is now being rolled out and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.
   * AKS Kubernetes patch versions 1.29.12, 1.29.13, 1.30.8, 1.30.9, 1.31.4, and 1.31.5 are now available.
```

   * 

* Preview Features:

>    * AKS Kubernetes version 1.32 is now in preview.

Reviewer 4: ```suggestion
```
@kaarthis did you also add this to the last release notes?.

>    * [Kubelet serving certificate rotation](https://learn.microsoft.com/Azure/AKS/certificate-rotation) now allows AKS to utilize kubelet server TLS bootstrapping for both bootstrapping and rotating serving certificates signed by the Cluster CA.

Reviewer 4: ```suggestion
```
This is only available in non-public regions. Not to be included in release notes until public

>    * [Azure Linux v3](https://learn.microsoft.com/azure/azure-linux/) is now in preview for AKS 1.32 clusters.

Reviewer 4: ```suggestion
```
@kaarthis has a change to last release notes to include the new AzLinux 3.0 default.

* Bug Fixes:

>    * Updated the Istio service mesh add-on to allow multiple operations with the Lua EnvoyFilter.

Reviewer 3: ```suggestion
   * Resolved an issue with Istio service mesh add-on where having multiple operations with the Lua EnvoyFilter was not allowed.
```

Can you put it in this format?
> Resolved an issue with [component] where [undesired behavior was happening].

Example:

> Resolved an issue with Istio service mesh add-on where having multiple operations with the Lua EnvoyFilter was not allowed.

   *  

* Behavior Changes:

>    * You can now configure the [default ingress controller type](https://learn.microsoft.com/en-us/azure/aks/app-routing) in the AKS application routing add-on.

Reviewer 3: ```suggestion

```

Removing this. This is in the Feature list.

>    * AKS no longer supports the [GPU image (preview)](https://github.com/Azure/AKS/issues/4472) to provision GPU-enabled AKS nodes. Alternative options that are supported today and recommended by AKS include the default experience with manual NVIDIA device plugin installation or the NVIDIA GPU Operator, detailed in [AKS GPU node pool documentation](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#confirm-that-gpus-are-schedulable).

Reviewer 4: ```suggestion
   * AKS no longer supports the [GPU image (preview)](https://github.com/Azure/AKS/issues/4472) to provision GPU-enabled AKS nodes. Any attempts to create new node pools with the GPU image (preview) will fail. Alternative options that are supported today and recommended by AKS include the default experience with manual NVIDIA device plugin installation or the NVIDIA GPU Operator, detailed in [AKS GPU node pool documentation](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#confirm-that-gpus-are-schedulable).
```

>    * AKS no longer supports the [GPU image (preview)](https://github.com/Azure/AKS/issues/4472) to provision GPU-enabled AKS nodes. Any attempts to create new node pools with the GPU image (preview) will fail. Alternative options that are supported today and recommended by AKS include the default experience with manual NVIDIA device plugin installation or the NVIDIA GPU Operator, detailed in [AKS GPU node pool documentation](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#confirm-that-gpus-are-schedulable).

Reviewer 3: ```suggestion
```
Remove from here and put in the announcements because this is a retirement.

   * Starting in the Karpenter v0.6 image, Node Auto Provisioning (NAP) will bootstrap nodes from Scriptless to AgentBaker using custom data provided by Node Provisioning Service (NPS) and improve the stability of NAP.

> * Component Updates:

Reviewer 4: Please include Windows, Ubuntu, and AzLinux vhd release notes according to the instructions here: https://msazure.visualstudio.com/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/21515/Writing-weekly-AKS-release-notes?anchor=vhd-images)

   * Beginning in Kubernetes 1.32, AKS uses CNS to install Azure CNI NodeSubnet by default, meaning users will now see the CNS daemonset in their environments with no changes to functionality or required action.

>    * Updated ExtensionManager image version to [v1.22.4](https://github.com/azure-core/ClusterConfigurationAgent/releases/tag/v1.22.4).

Reviewer 3: ```suggestion
   * ExtensionManager has been upgraded to  [v1.22.4](https://github.com/azure-core/ClusterConfigurationAgent/releases/tag/v1.22.4).
```

>    * For Node Auto Provisioning, the Azure provider of Karpenter is now updated to [v0.7.2](https://github.com/Azure/karpenter-provider-azure/releases/tag/v0.7.2).

Reviewer 3: ```suggestion
   * Node Auto Provisioning has been upgraded to use Karpenter [v0.7.2](https://github.com/Azure/karpenter-provider-azure/releases/tag/v0.7.2).
```

>    * Beginning in Kubernetes 1.32, AKS uses CNS to install Azure CNI NodeSubnet by default, meaning users will now see the CNS daemonset in their environments with no changes to functionality or required action.

Author: ```suggestion
   * Starting with Kubernetes v1.32, node subnet mode will be installed via a DaemonSet (CNS) instead of AgentBaker. Customers will now see this DaemonSet running in their clusters, aligning node subnet mode with PodSubnet and Overlay CNI for a more uniform architecture and faster security updates.
```

   * Updated LTS patch version 1.27.102 for Command Injection affecting Windows nodes to address [CVE-2024-9042](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-9042).
   * Updated the Retina basic image to v0.0.25 for Linux and Windows to address [CVE-2025-23047](https://nvd.nist.gov/vuln/detail/CVE-2025-23047) and [CVE-2024-45338](https://nvd.nist.gov/vuln/detail/CVE-2024-45338).
   * Updated the cost-analysis-agent image from v0.0.20 to v0.0.21. Upgrades the following dependencies in cost-analysis-agent to fix [CVE-2024-45341](https://nvd.nist.gov/vuln/detail/CVE-2024-45341) and [CVE-2024-45336](https://nvd.nist.gov/vuln/detail/CVE-2024-45336):
      * github.com/Azure/azure-sdk-for-go/sdk/azcore v1.15.0 to v1.17.0
      * github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.0 to v1.8.1
      * github.com/prometheus/common v0.60.0 to v0.62.0
      * github.com/samber/lo v1.47.0 to v1.49.1
      * github.com/stretchr/testify v1.9.0 to v1.10.0

>    * addon-resizer image v1.8.23-2 has been built using 
>     the Dalec framework, tested and published to MCR under [oss/v2 path](mcr.microsoft.com/oss/v2/kubernetes/autoscaler/addon-resizer:v1.8.23-2). All AKS clusters will use v1.8.23-2 addon-resizer image version.

Reviewer 3: ```suggestion
   * addon-resizer has been upgraded to v1.8.23-2.
```

Reviewer: is addon-resizer a user facing or publicly available component? if not, and if there's no direct behavior change, should we remove this?

Reviewer 2: Yes anything does not have proper Docs or cannot be searched in AKS pages pls feel free to remove them. As it causes more confusion than address anything.

   * Updated Azure File CSI driver liveness probe to v2.15.0 and CSI node driver registrar to v2.13.0 on AKS 1.31 and 1.32.
   * cluster-proportional-autoscaler (CPA) image v1.9.0-1 has been built using the Dalec framework, tested and published to MCR under [oss/v2 path](mcr.microsoft.com/oss/v2/kubernetes/autoscaler/cluster-proportional-autoscaler:v1.9.0-1). All AKS clusters with 1.27+ version will use v1.8.11-5 CPA image version.

> 

Author: * Extend the AKS security patch release notes in release tracker to include a package comparison with the current - 1 AKS Ubuntu base image.

>    * cluster-proportional-autoscaler (CPA) image v1.9.0-1 has been built using the Dalec framework, tested and published to MCR under [oss/v2 path](mcr.microsoft.com/oss/v2/kubernetes/autoscaler/cluster-proportional-autoscaler:v1.9.0-1). All AKS clusters with 1.27+ version will use v1.8.11-5 CPA image version.

Reviewer 3: ```suggestion
   * cluster-proportional-autoscaler has been upgraded to v1.9.0-1.
```

> 

Author: * Extend the AKS security patch release notes in release tracker to include a package comparison with the current - 1 AKS Ubuntu base image.
   * AKS Azure Linux v2 image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSAzureLinux/gen2/202502.09.0.txt).
   * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSUbuntu/gen2/2204containerd/202502.09.0.txt).
   * AKS Windows Server 2019 image has been updated to [17763.6775.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.6775.250117.txt).
   * AKS Windows Server 2022 image has been updated to [20348.3091.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd/20348.3091.250117.txt).
   * AKS Windows Server 23H2 image has been updated to [25398.1369.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/23H2/25398.1369.250117.txt)

> 

Author: ```suggestion
   * Extend the AKS security patch release notes in release tracker to include a package comparison with the current - 1 AKS Ubuntu base image.
   * AKS Azure Linux v2 image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSAzureLinux/gen2/202502.09.0.txt).
   * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSUbuntu/gen2/2204containerd/202502.09.0.txt).
   * AKS Windows Server 2019 image has been updated to [17763.6775.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.6775.250117.txt).
   * AKS Windows Server 2022 image has been updated to [20348.3091.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd/20348.3091.250117.txt).
   * AKS Windows Server 23H2 image has been updated to [25398.1369.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/23H2/25398.1369.250117.txt)
```

   * Updated the Retina basic image to v0.0.25 for Linux and Windows to address [CVE-2025-23047](https://nvd.nist.gov/vuln/detail/CVE-2025-23047) and [CVE-2024-45338](https://nvd.nist.gov/vuln/detail/CVE-2024-45338).
   * Updated the cost-analysis-agent image from v0.0.20 to v0.0.21. Upgrades the following dependencies in cost-analysis-agent to fix [CVE-2024-45341](https://nvd.nist.gov/vuln/detail/CVE-2024-45341) and [CVE-2024-45336](https://nvd.nist.gov/vuln/detail/CVE-2024-45336):
      * github.com/Azure/azure-sdk-for-go/sdk/azcore v1.15.0 to [v1.17.0](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/CHANGELOG.md#1170-2025-01-07)
      * github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.0 to [v1.8.1](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/CHANGELOG.md#181-2025-01-15)
      * github.com/prometheus/common v0.60.0 to [v0.62.0](https://github.com/prometheus/common/releases/tag/v0.62.0)
      * github.com/samber/lo v1.47.0 to [v1.49.1](https://github.com/samber/lo/releases/tag/v1.49.1)
      * github.com/stretchr/testify v1.9.0 to [v1.10.0](https://github.com/stretchr/testify/releases/tag/v1.10.0)

>    * AKS Azure Linux v2 image has been updated to [202502.09.0](https://github.com/Azure/AKS/blob/master/vhd-notes/AzureLinux/202502.09.0.txt).

Author: ```suggestion
   * AKS Azure Linux v2 image has been updated to [202502.09.0](vhd-notes/AzureLinux/202502.09.0.txt).
```

>    * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202502.09.0.txt).

Author: ```suggestion
   * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202502.09.0.txt).
   * AKS Ubuntu 24.04 image has been updated to [202502.09.0](vhd-notes/aks-ubuntu/AKSUbuntu-2404/202502.09.0.txt).
```

>    * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](https://github.com/Azure/AKS/blob/master/vhd-notes/aks-ubuntu/AKSUbuntu-2204/202502.09.0.txt).

Author: ```suggestion
   * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202502.09.0.txt).
```

>    * AKS Windows Server 2019 image has been updated to [17763.6775.250117](https://github.com/Azure/AKS/blob/master/vhd-notes/AKSWindows/2019/17763.6775.250117.txt).

Author: ```suggestion
   * AKS Windows Server 2019 image has been updated to [17763.6775.250117](vhd-notes/AKSWindows/2019/17763.6775.250117.txt).
```

>    * AKS Windows Server 2022 image has been updated to [20348.3091.250117](https://github.com/Azure/AKS/blob/master/vhd-notes/AKSWindows/2022/20348.3091.250117.txt).

Author: ```suggestion
   * AKS Windows Server 2022 image has been updated to [20348.3091.250117](vhd-notes/AKSWindows/2022/20348.3091.250117.txt).
```

# Azure Kubernetes Service Changelog

## Release 2025-02-20

Monitor the release status by region at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled `v20250220`.

### Announcements
* HTTP Application Routing (preview) is going to be [retired on March 3, 2025](https://azure.microsoft.com/updates?id=retirement-http-application-routing-addon-preview-for-aks-will-retire-03032025) and AKS will start to block new cluster creation with HTTP Application Routing (preview) enabled. Affected clusters must migrate to the generally available Application Routing add-on prior to that date. Refer to the [migration guide](https://learn.microsoft.com/en-us/azure/aks/app-routing-migration) for more information.

### Release Notes
* Features:
   * Application routing add-on support for [configuring the default NGINX ingress controller visibility](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration?tabs=azurecli#control-the-default-ingress-controller-configuration-when-creating-the-cluster) is now generally available in API 2025-02-01.

   * 

>    * AKS Kubernetes patch versions 1.29.12, 1.29.13, 1.30.8, 1.30.9, 1.31.4, and 1.31.5 are now available.

Reviewer 2: Lets put a link to the upstream patches page - https://kubernetes.io/releases/patch-releases/

   * 

* Preview Features:

>    * AKS Kubernetes version 1.32 is now being rolled out in preview and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.

Reviewer 2: This is not Preview but GA version and please put this in Announcement not preview section.

      * github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.0 to v1.8.1
      * github.com/prometheus/common v0.60.0 to v0.62.0
      * github.com/samber/lo v1.47.0 to v1.49.1
      * github.com/stretchr/testify v1.9.0 to v1.10.0
   * addon-resizer image v1.8.23-2 has been built using 
    the Dalec framework, tested and published to MCR under [oss/v2 path](mcr.microsoft.com/oss/v2/kubernetes/autoscaler/addon-resizer:v1.8.23-2). All AKS clusters will use v1.8.23-2 addon-resizer image version.
   * Updated Azure File CSI driver liveness probe to v2.15.0 and CSI node driver registrar to v2.13.0 on AKS 1.31 and 1.32.
   * cluster-proportional-autoscaler has been upgraded to v1.9.0-1.
   * Extend the AKS security patch release notes in release tracker to include a package comparison with the current - 1 AKS Ubuntu base image.
   * AKS Azure Linux v2 image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSAzureLinux/gen2/202502.09.0.txt).
   * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSUbuntu/gen2/2204containerd/202502.09.0.txt).

>    * AKS Windows Server 2019 image has been updated to [17763.6775.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.6775.250117.txt).

Reviewer 2: I would suggest removing all the Linux, Windows release info as they are completely orthogonal to the AKS Release train as each has its own toggle in different cadence. Pls put the OS link from release tracker so ppl can self serve - https://releases.aks.azure.com/webpage/index.html

Reviewer: let's discuss this internally as it's a departure from our existing process. for this release, I suggest we keep them

>    * cluster-proportional-autoscaler has been upgraded to v1.9.0-1.

Author: ```suggestion
```

# Azure Kubernetes Service Changelog

## Release 2025-02-20

Monitor the release status by region at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled `v20250220`.

### Announcements

> * AKS Kubernetes version 1.32 is now being rolled out and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.

Author: ```suggestion
* AKS Kubernetes version 1.32 is rolling out soon and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.
```

> * HTTP Application Routing (preview) is going to be [retired on March 3, 2025](https://azure.microsoft.com/updates?id=retirement-http-application-routing-addon-preview-for-aks-will-retire-03032025) and AKS will start to block new cluster creation with HTTP Application Routing (preview) enabled. Affected clusters must migrate to the generally available Application Routing add-on prior to that date. Refer to the [migration guide](https://learn.microsoft.com/en-us/azure/aks/app-routing-migration) for more information.

Reviewer: ```suggestion
* HTTP Application Routing (preview) is going to be [retired on March 3, 2025](https://azure.microsoft.com/updates?id=retirement-http-application-routing-addon-preview-for-aks-will-retire-03032025) and AKS will start to block new cluster creation with HTTP Application Routing (preview) enabled. Affected clusters must migrate to the generally available Application Routing add-on prior to that date. Refer to the [migration guide](https://learn.microsoft.com/azure/aks/app-routing-migration) for more information.
```

> * Using the GPU VHD image (preview) to provision GPU-enabled AKS nodes was retired on January 10, 2025 and AKS will block creation of new node pools with the GPU VHD image (preview). Follow the detailed steps to [create GPU-enabled node pools](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#use-the-aks-gpu-image-preview) using the alternative supported options.

Author: ```suggestion
* Using the GPU VHD image (preview) to provision GPU-enabled AKS nodes was retired on January 10, 2025 and AKS will block creation of new node pools with the GPU VHD image (preview). Follow the detailed steps to [create GPU-enabled node pools](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#use-the-aks-gpu-image-preview) using the alternative supported options.
* Extend the AKS [security patch release notes](https://releases.aks.azure.com/) in release tracker to include a package comparison with the current - 1 AKS Ubuntu base image.
```

### Release Notes
* Features:
   * Application routing add-on support for [configuring the default NGINX ingress controller visibility](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration?tabs=azurecli#control-the-default-ingress-controller-configuration-when-creating-the-cluster) is now generally available in API 2025-02-01.

>    * Kubernetes events for monitoring [node auto-repair](https://learn.microsoft.com/en-us/azure/aks/node-auto-repair) actions are now available for your AKS cluster. You can ingest these events and create alerts following the same [process as other Kubernetes events](https://learn.microsoft.com/azure/aks/events?tabs=azure-cli).

Reviewer: ```suggestion
   * Kubernetes events for monitoring [node auto-repair](https://learn.microsoft.com/azure/aks/node-auto-repair) actions are now available for your AKS cluster. You can ingest these events and create alerts following the same [process as other Kubernetes events](https://learn.microsoft.com/azure/aks/events?tabs=azure-cli).
```

   * 

* Preview Features:
   * AKS Kubernetes version 1.32 is now being rolled out in preview and is expected to reach all regions on or before the end of March. Please use the [az-aks-get-versions](https://learn.microsoft.com/azure/aks?view=azure-cli-latest#az-aks-get-versions) command to accurately capture if Kubernetes version 1.32 is available in your region.
   * You can use the `EnableCiliumNodeSubnet' feature in preview to [create Cilium node subnet clusters](https://learn.microsoft.com/azure/aks/azure-cni-powered-by-cilium#option-3-assign-ip-addresses-from-the-node-subnet-preview) using Azure CNI Powered by Cilium.

* Bug Fixes:
   * Resolved an issue with Istio service mesh add-on where having multiple operations with the Lua EnvoyFilter was not allowed.

>    *  Fixed a bug in Azure CNI Pod Subnet Static Block Allocation mode with Cilium which caused incorrect iptables rules, leading to pod connectivity failures to DNS and IMDS.

Author: ```suggestion
   * Fixed a bug in Azure CNI Pod Subnet Static Block Allocation mode with Cilium which caused incorrect iptables rules, leading to pod connectivity failures to DNS and IMDS.
   * Resolved an [issue](https://github.com/Azure/azure-sdk-for-go/issues/24109) where the updated Azure Table client mishandled untyped numbers, causing static block agent pools to be misidentified as dynamic and leading to operation failures.
   * Fixed a bug in Azure Kubernetes Fleet Manager hubful resource groups by truncating the name to avoid issues with long generated managed resource group names breaking the maximum length of resource groups.
```

>    * You can use the `EnableCiliumNodeSubnet' feature in preview to [create Cilium node subnet clusters](https://learn.microsoft.com/azure/aks/azure-cni-powered-by-cilium#option-3-assign-ip-addresses-from-the-node-subnet-preview) using Azure CNI Powered by Cilium.

Author: ```suggestion
   * You can use the `EnableCiliumNodeSubnet` feature in preview to [create Cilium node subnet clusters](https://learn.microsoft.com/azure/aks/azure-cni-powered-by-cilium#option-3-assign-ip-addresses-from-the-node-subnet-preview) using Azure CNI Powered by Cilium.
```

>    * You can use the `EnableCiliumNodeSubnet' feature in preview to [create Cilium node subnet clusters](https://learn.microsoft.com/azure/aks/azure-cni-powered-by-cilium#option-3-assign-ip-addresses-from-the-node-subnet-preview) using Azure CNI Powered by Cilium.

Reviewer: nit: there's ` before EnableCiliumNodeSubnet and ' after

>    * [Free control plane metrics](https://learn.microsoft.com/azure/aks/control-plane-metrics-monitor) are now available in preview to monitor the API server and ETCD components, which will be automatically collected by default on your cluster.

Reviewer: ```suggestion
   * [Control plane metrics](https://learn.microsoft.com/azure/aks/control-plane-metrics-monitor) are now available through Azure Monitor platform metrics in preview to monitor critical control plane components such as API server, etcd, scheculer, autoscaler, and controller-manager.

>    * Resolved an [issue](https://github.com/Azure/azure-sdk-for-go/issues/24109) where the updated Azure Table client mishandled untyped numbers, causing static block agent pools to be misidentified as dynamic and leading to operation failures.

Reviewer: is this user facing? what's 'static block' agent pools?

Reviewer: Ah it might be referring to azure cni static block allocation mode. if yes, may be good to reference that to clarify on static block and dymanic references. Also, would be good to replace 'agent pools' with 'node pools'

Author: Ack, updated

>    * Resolved an [issue](https://github.com/Azure/azure-sdk-for-go/issues/24109) where the updated Azure Table client mishandled untyped numbers, causing static block agent pools to be misidentified as dynamic and leading to operation failures.

Author: ```suggestion
   * Resolved an [issue](https://github.com/Azure/azure-sdk-for-go/issues/24109) in Azure CNI static block IP allocation mode, where the updated Azure Table client mishandled untyped numbers, causing static block node pools to be misidentified as dynamic and leading to operation failures.
```

>    * Fixed a bug in Azure Kubernetes Fleet Manager hubful resource groups by truncating the name to avoid issues with long generated managed resource group names breaking the maximum length of resource groups.

Reviewer: ```suggestion
   * Fixed a bug in Azure Kubernetes Fleet Manager hub cluster resource groups (FL_ prefix resource groups) by truncating the name to avoid issues with long generated managed resource group names breaking the maximum length of resource groups.
```

   * Updated Windows CNS from v1.6.13 to v1.6.21 and Linux CNS from v1.6.18 to v1.6.21.
   * Updated Windows CNI and Linux CNI from v1.6.18 to v1.6.21.
   * Updated [tigera operator to v1.36.3](https://github.com/tigera/operator/releases/tag/v1.36.3) and [calico to v3.29.0](https://github.com/projectcalico/calico/releases/tag/v3.29.0).
   * Node Auto Provisioning has been upgraded to use Karpenter [v0.7.2](https://github.com/Azure/karpenter-provider-azure/releases/tag/v0.7.2).
   * Updated LTS patch version 1.27.102 for Command Injection affecting Windows nodes to address [CVE-2024-9042](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-9042).
   * Updated the Retina basic image to v0.0.25 for Linux and Windows to address [CVE-2025-23047](https://nvd.nist.gov/vuln/detail/CVE-2025-23047) and [CVE-2024-45338](https://nvd.nist.gov/vuln/detail/CVE-2024-45338).
   * Updated the cost-analysis-agent image from v0.0.20 to v0.0.21. Upgrades the following dependencies in cost-analysis-agent to fix [CVE-2024-45341](https://nvd.nist.gov/vuln/detail/CVE-2024-45341) and [CVE-2024-45336](https://nvd.nist.gov/vuln/detail/CVE-2024-45336):

>       * github.com/Azure/azure-sdk-for-go/sdk/azcore v1.15.0 to v1.17.0

Author: ```suggestion
      * github.com/Azure/azure-sdk-for-go/sdk/azcore v1.15.0 to [v1.17.0](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/CHANGELOG.md#1170-2025-01-07)
```

>       * github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.0 to v1.8.1

Author: ```suggestion
      * github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.0 to [v1.8.1](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/CHANGELOG.md#181-2025-01-15)
```

>       * github.com/prometheus/common v0.60.0 to v0.62.0

Author: ```suggestion
      * github.com/prometheus/common v0.60.0 to [v0.62.0](https://github.com/prometheus/common/releases/tag/v0.62.0)
```

>       * github.com/samber/lo v1.47.0 to v1.49.1

Author: ```suggestion
      * github.com/samber/lo v1.47.0 to [v1.49.1](https://github.com/samber/lo/releases/tag/v1.49.1)
```

>       * github.com/stretchr/testify v1.9.0 to v1.10.0

Author: ```suggestion
      * github.com/stretchr/testify v1.9.0 to [v1.10.0](https://github.com/stretchr/testify/releases/tag/v1.10.0)
```

>    * [Horizontal Pod Autoscaling](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) introduced for `ama-metrics replicaset pod` in the [Azure Monitor managed service for Prometheus](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/prometheus-metrics-overview) add-on. More details about the configuration of the Horizontal Pod Autoscaler can be found [here](https://github.com/Azure/prometheus-collector/blob/main/internal/docs/HPA.md).

Author: ```suggestion
   * [Horizontal Pod Autoscaling](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) introduced for `ama-metrics replicaset pod` in the [Azure Monitor managed service for Prometheus](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/prometheus-metrics-overview) add-on. More details about the configuration of the Horizontal Pod Autoscaler can be found [here](https://github.com/Azure/prometheus-collector/blob/main/internal/docs/HPA.md).
   * Starting with Kubernetes v1.32, node subnet mode will be installed via the `azure-cns` DaemonSet, allowing for faster security updates.
```

>    * Starting with Kubernetes v1.32, node subnet mode will be installed via a DaemonSet (CNS) instead of AgentBaker. Customers will now see this DaemonSet running in their clusters, aligning node subnet mode with PodSubnet and Overlay CNI for a more uniform architecture and faster security updates.

Reviewer: need to remove 'AgentBaker'

Reviewer: can we mention the exact name of the daemonset? iirc it should be azure-cns. if yes, then it should be "Starting with Kubernetes v1.32, node subnet mode will be installed via `azure-cns` DaemonSet, allowing for faster security updates"

Reviewer: also this may need to be under behavior change section. need to see the PR, will sync offline

>    * Updated Windows CNI and Linux CNI from v1.6.18 to v1.6.21.

Reviewer: - these seem to be duplicate lines covering same updates
- can we add links to the releases from here (https://github.com/Azure/azure-container-networking/releases)?

Author: Updated the links

>    * Updated Windows CNS from v1.6.13 to v1.6.21 and Linux CNS from v1.6.18 to v1.6.21.

Author: ```suggestion
   * Updated Windows CNS from v1.6.13 to [v1.6.21](https://github.com/Azure/azure-container-networking/releases/tag/v1.6.21) and Linux CNS from v1.6.18 to [v1.6.21](https://github.com/Azure/azure-container-networking/releases/tag/v1.6.21).
```

>    * Updated Windows CNI and Linux CNI from v1.6.18 to v1.6.21.

Author: ```suggestion
   * Updated Windows CNI and Linux CNI from v1.6.18 to [v1.6.21](https://github.com/Azure/azure-container-networking/releases/tag/v1.6.21).
```

>    * Starting with Kubernetes v1.32, node subnet mode will be installed via a DaemonSet (CNS) instead of AgentBaker. Customers will now see this DaemonSet running in their clusters, aligning node subnet mode with PodSubnet and Overlay CNI for a more uniform architecture and faster security updates.

Author: ```suggestion
```

      * github.com/stretchr/testify v1.9.0 to [v1.10.0](https://github.com/stretchr/testify/releases/tag/v1.10.0)

>    * Updated Azure File CSI driver liveness probe to v2.15.0 and CSI node driver registrar to v2.13.0 on AKS 1.31 and 1.32.

Author: ```suggestion
```

>    * Extend the AKS [security patch release notes](https://releases.aks.azure.com/) in release tracker to include a package comparison with the current - 1 AKS Ubuntu base image.

Reviewer: this should go under announcements section

>    * Extend the AKS [security patch release notes](https://releases.aks.azure.com/) in release tracker to include a package comparison with the current - 1 AKS Ubuntu base image.

Author: ```suggestion
```

>    * AKS Azure Linux v2 image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSAzureLinux/gen2/202502.09.0.txt).

Author: ```suggestion
   * AKS Azure Linux v2 image has been updated to [202502.09.0](https://github.com/Azure/AKS/blob/master/vhd-notes/AzureLinux/202502.09.0.txt).
```

>    * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSUbuntu/gen2/2204containerd/202502.09.0.txt).

Author: ```suggestion
   * AKS Ubuntu 22.04 node image has been updated to [202502.09.0](https://github.com/Azure/AKS/blob/master/vhd-notes/aks-ubuntu/AKSUbuntu-2204/202502.09.0.txt).
```

>    * AKS Windows Server 2019 image has been updated to [17763.6775.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.6775.250117.txt).

Author: ```suggestion
   * AKS Windows Server 2019 image has been updated to [17763.6775.250117](https://github.com/sdesai345/AKS/blob/master/vhd-notes/AKSWindows/2019/17763.6775.250117.txt).
```

>    * AKS Windows Server 2022 image has been updated to [20348.3091.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd/20348.3091.250117.txt).

Author: ```suggestion
   * AKS Windows Server 2022 image has been updated to [20348.3091.250117](https://github.com/Azure/AKS/blob/master/vhd-notes/AKSWindows/2022/20348.3091.250117.txt).
```

>    * AKS Windows Server 23H2 image has been updated to [25398.1369.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/23H2/25398.1369.250117.txt).

Author: ```suggestion
   * AKS Windows Server 23H2 image has been updated to [25398.1369.250117](https://github.com/Azure/AKS/blob/master/vhd-notes/AKSWindows/23H2/25398.1369.250117.txt).
```

>    * AKS Windows Server 23H2 image has been updated to [25398.1369.250117](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/23H2/25398.1369.250117.txt).

Reviewer: we need to copy these files into vhd-notes folder and then reference them (similar to previous releases)

Author: Ack

* HTTP Application Routing (preview) is going to be [retired on March 3, 2025](https://azure.microsoft.com/updates?id=retirement-http-application-routing-addon-preview-for-aks-will-retire-03032025) and AKS will start to block new cluster creation with HTTP Application Routing (preview) enabled. Affected clusters must migrate to the generally available Application Routing add-on prior to that date. Refer to the [migration guide](https://learn.microsoft.com/en-us/azure/aks/app-routing-migration) for more information.

### Release Notes
* Features:
   * Application routing add-on support for [configuring the default NGINX ingress controller visibility](https://learn.microsoft.com/azure/aks/app-routing-nginx-configuration?tabs=azurecli#control-the-default-ingress-controller-configuration-when-creating-the-cluster) is now generally available in API 2025-02-01.
   * Kubernetes events for monitoring [node auto-repair](https://learn.microsoft.com/en-us/azure/aks/node-auto-repair) actions are now available for your AKS cluster. You can ingest these events and create alerts following the same [process as other Kubernetes events](https://learn.microsoft.com/azure/aks/events?tabs=azure-cli).

>    * 

Author: ```suggestion
```

   * AKS [Kubernetes patch versions](https://kubernetes.io/releases/patch-releases/) 1.29.12, 1.29.13, 1.30.8, 1.30.9, 1.31.4, and 1.31.5 are now available.

>    * 

Author: ```suggestion
```

> 

Author: ```suggestion
```

   * [Free control plane metrics](https://learn.microsoft.com/azure/aks/control-plane-metrics-monitor) are now available in preview to monitor the API server and ETCD components, which will be automatically collected by default on your cluster.

* Bug Fixes:
   * Resolved an issue with Istio service mesh add-on where having multiple operations with the Lua EnvoyFilter was not allowed.
   * Fixed a bug in Azure CNI Pod Subnet Static Block Allocation mode with Cilium which caused incorrect iptables rules, leading to pod connectivity failures to DNS and IMDS.
   * Resolved an [issue](https://github.com/Azure/azure-sdk-for-go/issues/24109) where the updated Azure Table client mishandled untyped numbers, causing static block agent pools to be misidentified as dynamic and leading to operation failures.
   * Fixed a bug in Azure Kubernetes Fleet Manager hubful resource groups by truncating the name to avoid issues with long generated managed resource group names breaking the maximum length of resource groups.

> 

Author: ```suggestion
```

  * Security patch releases in release tracker, starting with 20250115T000000Z will contain release notes for the release.

> * Preview Features:

Reviewer 5: Add support for Node Subnet IPAM mode in AKS clusters using Azure CNI ( by Cilium) in the preview list. @sf-msft
